current_spring2026/graph/graph_builder.py

```
@@ -0,0 +1,244 @@
+                chunk_data.append((doc, entities))
+
+        print(f"  Extracted entities from {len(chunk_data)} docs")
+
+        if not chunk_data:
+            continue
+
+        # ── Phase 2: Embed all entities in one GPU batch ───────────────────
+        all_texts = [build_entity_text(e) for _, ents in chunk_data for e in ents]
+        print(f"  Embedding {len(all_texts)} entities on GPU...")
+        all_embs  = embedder.embed(all_texts)
+        print(f"  Embedding complete")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)